### PR TITLE
Handle unset experiment in log_artifact(s) call

### PR DESCRIPTION
**Pull request by rcarmstrong**
Not entirely sure why self.experiment is set to None in the Experiment class __init__, regardless I've added a simple if statement that will allow the use of the log_artifact(s) helper methods and shouldn't impact existing functionality.

### CODE REVIEW

File: polyaxon_client/tracking/experiment.py [hunk 1]
```diff
@@ -217,11 +217,17 @@ def log_framework(self, framework):
 
     @check_no_op
     def log_artifact(self, file_path):
```
Comment (mouradmourafiq) [inline polyaxon_client/tracking/experiment.py]: I am not sure what I was trying to do here, but this is definitely a wrong implementation, this should be simply:

```python
    @check_no_op
    def log_artifact(self, file_path):
        self.outputs_store.upload_file(file_path)

    @check_no_op
    def log_artifacts(self, dir_path):
        self.outputs_store.upload_dir(dirname=dirname)
```

**Comment by mouradmourafiq** [comment]
I think the implementation was just wrong, there are also 2 other methods, `log_output` and `log_outputs` on the base tracker, I am going to clean the client before the v0.5 release, especially these artifacts method, since the platform will be providing annotation for `image`, `dataframes`, `model` among others to specify the type of the artifacts.

**Comment by rcarmstrong** [comment]
Is the v0.5 release happening soon? Should I close this?

**Comment by mouradmourafiq** [comment]
@rcarmstrong yes we are testing and pushing hard to have a RC soon.

**Comment by mouradmourafiq** [comment]
Just an update, I fixed the [docs](https://docs.polyaxon.com/references/polyaxon-tracking-api/experiments/#log_output) to reference the `log_output(s)` methods and fixed the `log_artifact(s)` or next release since the other method are deprecated.

**Pull request closed without merging**